src/bank.rs

```diff
@@ -1026,28 +1034,70 @@ impl Bank {
         // which implies its id can be used as the ledger's seed.
         let entry0 = entries.next().expect("invalid ledger: empty");
 
-        // The second item in the ledger is a special transaction where the to and from
+        // The second item in the ledger consists of a transaction with
+        // two special instructions:
+        // 1) The first is a special move instruction where the to and from
         // fields are the same. That entry should be treated as a deposit, not a
         // transfer to oneself.
+        // 2) The second is a move instruction that acts as a payment to the first
+        // leader from the mint. This bootstrap leader will stay in power during the
+        // bootstrapping period of the network
         let entry1 = entries
             .next()
             .expect("invalid ledger: need at least 2 entries");
         {
+            // Process the first transaction
             let tx = &entry1.transactions[0];
             assert!(SystemProgram::check_id(tx.program_id(0)), "Invalid ledger");
-            let instruction: SystemProgram = deserialize(tx.userdata(0)).unwrap();
-            let deposit = if let SystemProgram::Move { tokens } = instruction {
+            assert!(SystemProgram::check_id(tx.program_id(1)), "Invalid ledger");
+            let mut instruction: SystemProgram = deserialize(tx.userdata(0)).unwrap();
+            let mint_deposit = if let SystemProgram::Move { tokens } = instruction {
+                Some(tokens)
+            } else {
+                None
+            }.expect("invalid ledger, needs to start with a contract");
+
+            instruction = deserialize(tx.userdata(1)).unwrap();
+            let leader_payment = if let SystemProgram::Move { tokens } = instruction {
                 Some(tokens)
             } else {
                 None
             }.expect("invalid ledger, needs to start with a contract");
```

Comment:
let's update this error message: "invalid ledger, bootstrap leader payment expected" perhaps